### PR TITLE
K8SPS-43 Add replication error check to cr status

### PR DESCRIPTION
**Pull request by nonemax**
[K8SPS-43](https://jira.percona.com/browse/K8SPS-43) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/4ef1e8e/auto-config.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/4ef1e8e/config.log) |
| init-deploy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/4ef1e8e/init-deploy.log) |
| monitoring | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/4ef1e8e/monitoring.log) |
| semi-sync | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/4ef1e8e/semi-sync.log) |
| service-per-pod | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/4ef1e8e/service-per-pod.log) |
| scaling | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/4ef1e8e/scaling.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/4ef1e8e/sidecars.log) |
| users | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/4ef1e8e/users.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/4ef1e8e/limits.log) |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/4ef1e8e/demand-backup.log) |
| gr-init-deploy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/4ef1e8e/gr-init-deploy.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/148/commits/4ef1e8e26da2b66e278a6da1134304e9b94abd7d
image: `perconalab/percona-server-mysql-operator:PR-148-4ef1e8e`

File: api/v1alpha1/perconaservermysql_types.go [hunk 1]
```diff
@@ -275,8 +275,9 @@ type ServiceExposeTogglable struct {
 type StatefulAppState string
 
 const (
-	StateInitializing StatefulAppState = "initializing"
-	StateReady        StatefulAppState = "ready"
+	StateInitializing     StatefulAppState = "initializing"
+	StateReady            StatefulAppState = "ready"
+	StateReplicationError StatefulAppState = "replication error"
```
Comment (inelpandzic) [inline api/v1alpha1/perconaservermysql_types.go:280]: Can we have just a replication error here, what about other errors if any? Should we have a more general error in here if there are more things that can break besides replication?

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/2b4bf05/auto-config.log) |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/2b4bf05/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/2b4bf05/init-deploy.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/2b4bf05/config.log) |
| scaling | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/2b4bf05/scaling.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/2b4bf05/gr-init-deploy.log) |
| monitoring | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/2b4bf05/monitoring.log) |
| users | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/2b4bf05/users.log) |
| semi-sync | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/2b4bf05/semi-sync.log) |
| service-per-pod | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/2b4bf05/service-per-pod.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/2b4bf05/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/2b4bf05/limits.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/148/commits/2b4bf05432e1b443bb6304efe1f7a4083681d02f
image: `perconalab/percona-server-mysql-operator:PR-148-2b4bf05`

File: pkg/orchestrator/client.go [hunk 2]
```diff
@@ -177,3 +177,19 @@ func doRequest(ctx context.Context, url string) (*http.Response, error) {
 
 	return resp, nil
 }
+
+func GetReplicationStatus(ctx context.Context, apiHost string, clusterName string) (orcResp *OrcResponse, err error) {
```
Comment (egegunes) [inline pkg/orchestrator/client.go:181]: copying @inelpandzic comment about named returns:
> I suggest staying away from named return types, they introduce a bunch of janky things. To see those its best to see Dave Cheney tip on them: https://dave.cheney.net/practical-go/presentations/gophercon-israel.html#_avoid_named_return_values

File: api/v1alpha1/perconaservermysql_types.go (hunk 1 above)
Comment (egegunes) [reply api/v1alpha1/perconaservermysql_types.go:280]: @nonemax just "error" might be better, since we have the same for PSMDB and PXC operators

**Review by egegunes — CHANGES_REQUESTED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/566996e/auto-config.log) |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/566996e/demand-backup.log) |
| init-deploy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/566996e/init-deploy.log) |
| scaling | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/566996e/scaling.log) |
| gr-init-deploy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/566996e/gr-init-deploy.log) |
| users | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/566996e/users.log) |
| monitoring | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/566996e/monitoring.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/566996e/config.log) |
| semi-sync | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/566996e/semi-sync.log) |
| service-per-pod | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/566996e/service-per-pod.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/566996e/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/566996e/limits.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-148/566996e/tls-cert-manager.log) |
| We run 13 out of 13|

commit: https://github.com/percona/percona-server-mysql-operator/pull/148/commits/566996e0123e3601d231fd0edca765f5540add4b
image: `perconalab/percona-server-mysql-operator:PR-148-566996e`

**Pull request closed without merging**